src/app/qgsattributetabledialog.cpp

```diff
@@ -512,6 +512,11 @@ void QgsAttributeTableDialog::runFieldCalculation( QgsVectorLayer* layer, const
   }
 
   mLayer->endEditCommand();
+
+  // refresh table with updated values
+  // fixes https://issues.qgis.org/issues/17312
+  QgsAttributeTableModel* masterModel = mMainView->masterModel();
+  masterModel->reload( masterModel->index( 0, 0 ), masterModel->index( masterModel->rowCount() - 1, masterModel->columnCount() - 1 ) );
```

Comment:
Can you kill the `return` above and put this in an else?
Is it possible to only reload the affected column (I guess this reloads everything)?

Reply:
yes, that's was my first solution, but I wasn't able to find the correct api to get the involved column from its name.
my first solution was gettin the column with:
int modelColumn = masterModel->fieldIdx( fieldindex );
Is it the correct way?

Reply:
```cpp
    /**                                                                         
     * get field index from column                                              
     */                                                                         
    int fieldIdx( int col ) const;                                              
                                                                                
    /**                                                                         
     * get column from field index                                              
     */                                                                         
    int fieldCol( int idx ) const;
```